keystorage/hashicorp/src/main/java/tech/pegasys/signing/hashicorp/HashicorpConnectionFactory.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.signing.hashicorp;
+
+import tech.pegasys.signing.hashicorp.config.ConnectionParameters;
+import tech.pegasys.signing.hashicorp.config.TlsOptions;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpClientOptions;
+import io.vertx.core.net.JksOptions;
+import io.vertx.core.net.PemTrustOptions;
+import io.vertx.core.net.PfxOptions;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.net.tls.VertxTrustOptions;
+
+public class HashicorpConnectionFactory {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private final Vertx vertx;
+
+  public static final Long DEFAULT_TIMEOUT_MILLISECONDS = 10_000L;
+  public static final Integer DEFAULT_SERVER_PORT = 8200;
+
+  public HashicorpConnectionFactory(final Vertx vertx) {
+    this.vertx = vertx;
+  }
+
+  public HashicorpConnection create(final ConnectionParameters connectionParameters) {
+    final int serverPort = connectionParameters.getServerPort().orElse(DEFAULT_SERVER_PORT);
+
+    final HttpClientOptions httpClientOptions =
+        new HttpClientOptions()
+            .setDefaultHost(connectionParameters.getServerHost())
+            .setDefaultPort(serverPort);
+
+    final HttpClient httpClient;
+    try {
+      if (connectionParameters.getTlsOptions().isPresent()) {
+        LOG.debug("Connection to hashicorp vault using TLS.");
+        setTlsOptions(httpClientOptions, connectionParameters.getTlsOptions().get());
+      }
+      httpClient = vertx.createHttpClient(httpClientOptions);
```

Comment:
could just do the return HashicorpConnection here and avoid the need to declare the httpClient before the try